tapiro/core/src/main/scala/io/buildo/tapiro/AkkaHttpMeta.scala

```diff
@@ -0,0 +1,61 @@
+package io.buildo.tapiro
+
+import io.buildo.metarpheus.core.intermediate.Route
+
+import scala.meta._
+
+object AkkaHttpMeta {
+  val `class` = (
+    `package`: Term.Ref,
+    imports: Set[Term.Ref],
+    controllerName: Type.Name,
+    endpointsName: Term.Name,
+    implicits: List[Term.Param],
+    http4sEndpoints: List[Defn.Val],
+    routes: Term,
+  ) => {
+    val tapirEndpoints = q"val endpoints = $endpointsName.create()"
+    val akkaHttpEndpointsName = Term.Name(s"${controllerName.syntax}AkkaHttpEndpoints")
+    q"""
+    package ${`package`} {
+      ..${imports.toList.map(i => q"import $i._")}
+      import cats.effect._
+      import cats.implicits._
+      import cats.data.NonEmptyList
+      import sttp.tapir.server.akkahttp._
+      import sttp.tapir.Codec.{ JsonCodec, PlainCodec }
+      import akka.http.scaladsl.server._
+      import akka.http.scaladsl.server.Directives._
+
+      object $akkaHttpEndpointsName {
+        def routes(controller: $controllerName)(..$implicits): Route = {
+          ..${tapirEndpoints +: http4sEndpoints :+ routes}
+        }
+      }
+    }
+    """
+  }
+
+  val routes = (head: Route, tail: List[Route]) => {
+    val first = Term.Name(head.name.last)
+    val rest = tail.map(a => Term.Name(a.name.last))
+    q"List(..$rest).foldLeft[Route]($first)(_ ~ _)"
+  }
+
+  val endpoints = (routes: List[Route]) =>
```

Comment:
we might want to generalize this in a common module... but I think we're going to deprecate akkahttp and futures as soon as tagless final migration is done in gdsm and the other project